main.go

```diff
@@ -108,15 +112,55 @@ func init() {
 	prometheus.MustRegister(version.NewCollector("blackbox_exporter"))
 }
 
+func probeRequestHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
You're still arbitrarily splitting logic across probeRequestHandler and probeHandler. All this logic should be in one function, with a small shim anonymous function in the HandleFunc.